components/backup/src/endpoint.rs

```diff
@@ -769,18 +783,26 @@ impl<R: RegionInfoProvider> Progress<R> {
                             break;
                         }
                     }
-                    if info.role == StateRole::Leader {
+                    let peer = find_peer(region, store_id).unwrap().to_owned();
+                    // Raft peer role has to match the replica read flag.
+                    if (replica_read == BackupReplicaRead::Leader && info.role == StateRole::Leader)
+                        || (replica_read == BackupReplicaRead::Follower
+                            && info.role != StateRole::Leader)
```

Comment:
Need we check it so strictly? In other words, is it really necessary to distinguish between `BackupReplicaRead::Follower` and `BackupReplicaRead::LeaderAndFollower`?
If a peer just becomes leader but it receives a follower read request, I think we needn't reject it in this case.

Reply:
In most cases, `BackupReplicaRead::Follower`, `BackupReplicaRead::LeaderAndFollower` means nearly the same to the user. But, in some cases, for example, a user prefers all the leaders are in certain zones, while they also want the backup to run in other zones, then these 2 options make difference.

Reply:
> But, in some cases, for example, a user prefers all the leaders are in certain zones, while they also want the backup to run in other zones, then these 2 options make difference.

To my understanding, the user is perfectly clear about the zones where the TiKV stores are. However, leaders are less stable than stores themselves. Although we try to put all leaders in a certain zone, temporary network isolation or server crash may cause the leader to move to other zones.

So, in this case, the user can easily guarantee they don't send requests to a remote zone, but they cannot 100% guarantee the roles of the peers in the zone.

Reply:
I removed `FOLLOWER` only option